ext/bcmath/bcmath.c

```diff
@@ -1298,10 +1311,18 @@ static int bcmath_number_compare(zval *op1, zval *op2)
 static zend_always_inline zend_result bc_num_from_obj_or_str_or_long_with_err(
 	bc_num *num, size_t *scale, zend_object *obj, zend_string *str, zend_long lval, uint32_t arg_num)
 {
-	if (UNEXPECTED(bc_num_from_obj_or_str_or_long(num, scale, obj, str, lval) == FAILURE)) {
+	size_t full_scale = 0;
```

Comment:
I think it's okay to pass `scale` to `bc_num_from_obj_or_str_or_long` instead of `full_scale`, that allows a tiny bit of simplification.

Reply:
@nielsdos 

Since powmod does not use scale, pass NULL, which is why do this.

We could simplify this function by always passing in a non-NULL value in the call side, but I'm not sure which is preferable.

What do you think?

Reply:
I understand, then this approach is fine.